newsroom/push/publishing.py

```diff
@@ -298,3 +300,27 @@ async def publish_planning_into_event(self, planning: dict[str, Any]) -> Optiona
             # Replace the original document
             await service.update(agenda["_id"], agenda)
             return agenda["_id"]
+
+    async def publish_planning_related_events(self, planning: dict[str, Any], related_event: dict[str, Any]):
+        if not related_event.get("link_type") == "secondary":
+            return None
+
+        service = AgendaItemService()
+        event_id = related_event.get("_id")
+
+        event = await service.find_by_id(event_id)
+        agenda = event.to_dict()
+
+        if agenda.get("item_type") == "event" and planning and planning.get("guid") not in agenda["planning_ids"]:
+            agenda["planning_ids"].append(planning["guid"])
+
+        if event_id not in agenda["event_ids"]:
+            agenda["event_ids"].append(event_id)
+
+        planning_item = await service.find_by_id(planning["guid"])
+        if planning_item:
+            agenda["coverages"] = planning_item.coverages
```

Comment:
The Event might already have coverages from a previous push, from a different Planning item. This would then wipe those others off the Event.

Instead you will need to add the coverages from this Planning item to the Event, making sure not to include duplicates (based on the coverage_id field).